tst/schedulers/test_grid_searcher.py

```diff
@@ -106,3 +113,132 @@ def test_store_and_restore_state_with_initial_config():
     for idx in [3, 1, 2]:
         new_config = new_searcher.get_config(trail_id=idx)
         assert new_config == all_candidates_on_grid[idx]
+
+def test_grid_scheduler():
+    max_steps = 100
+    num_workers = 2
+    random_seed = 382378624
+
+    searcher = "grid"
+    mode = "min"
+    config_space = {
+        "steps": max_steps,
+        "width": randint(0, 20),
+        "height": uniform(-100, 100),
+        "sleep_time": 0.001,
+    }
+
+    num_samples = {"width": 5, "height": 10}
+    metric = "mean_loss"
+
+    entry_point = (
+        Path(__file__).parent.parent.parent
+        / "examples"
+        / "training_scripts"
+        / "height_example"
+        / "train_height.py"
+    )
+    trial_backend = LocalBackend(entry_point=str(entry_point))
+
+    search_options = {
+        "debug_log": True,
+        "num_init_random": num_workers,
+        "num_samples": num_samples,
+    }
+
+    myscheduler = FIFOScheduler(
+        config_space,
+        searcher=searcher,
+        search_options=search_options,
+        mode=mode,
+        metric=metric,
+        random_seed=random_seed,
+    )
+    stop_criterion = StoppingCriterion(max_wallclock_time=60)
+
+    tuner = Tuner(
+        trial_backend=trial_backend,
+        scheduler=myscheduler,
+        sleep_time=0.1,
+        n_workers=num_workers,
+        stop_criterion=stop_criterion,
+    )
+
+    tuner.run()
+
+
+def test_grid_scheduler_categorical():
```

Comment:
What's the difference of this unit test to the previous one `test_grid_scheduler()`

Reply:
test_grid_scheduler_categorical() tests grid searcher only for categorical variables, and in the absence of num_samples argument. test_grid_scheduler() tests for continuous variables. 
You are right there is alot of overlap, i will keep one of them only.

Reply:
Makes sense. Could you rename test_grid_scheduler() to make this clear? Or merge them together?